contracts/schemes/FundingRequest.sol

```diff
@@ -0,0 +1,164 @@
+pragma solidity ^0.5.16;
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "../libs/StringUtil.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "./CommonInterface.sol";
+
+
+/**
+ * @title A scheme for requesting funding from an organization
+ */
+
+contract FundingRequest is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable,
+        CommonInterface {
+    using SafeMath for uint;
+    using StringUtil for string;
+
+    event NewFundingProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address payable _beneficiary,
+        uint256 _amount,
+        string _descriptionHash
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, bool _decision);
+
+    event Redeem(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount
+    );
+
+    struct Proposal {
+        address payable beneficiary;
+        uint256 amount;
+        string descriptionHash;
+        uint256 executionTime;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    address public fundingToken;
```

Comment:
maybe type should be IERC20 . see [here](https://github.com/daostack/arc/blob/arc-factory/contracts/schemes/JoinAndQuit.sol#L67)